processing/src/test/java/org/apache/druid/query/topn/TopNQueryRunnerTest.java

```diff
@@ -810,6 +812,117 @@ public void testTopNOverHyperUniqueExpressionRounded()
     assertExpectedResults(expectedResults, query);
   }
 
+  @Test
+  public void testTopNOverStringFirstLastAggregator()
+  {
+    TopNQuery query = new TopNQueryBuilder()
+        .dataSource(QueryRunnerTestHelper.dataSource)
+        .granularity(QueryRunnerTestHelper.monthGran)
+        .dimension(QueryRunnerTestHelper.marketDimension)
+        .metric("last")
+        .threshold(3)
+        .intervals(QueryRunnerTestHelper.fullOnIntervalSpec)
+        .aggregators(
+            Arrays.asList(
+                new StringFirstAggregatorFactory("first", "qualityNumericString", null),
+                new StringLastAggregatorFactory("last", "qualityNumericString", null)
+            )
+        )
+        .build();
+
+    List<Result<TopNResultValue>> expectedResults = Arrays.asList(
+        new Result<>(
+            DateTimes.of("2011-01-01T00:00:00.000Z"),
+            new TopNResultValue(
+                Arrays.<Map<String, Object>>asList(
+                    ImmutableMap.<String, Object>builder()
+                        .put("market", "spot")
+                        .put("first", "100000")
+                        .put("last", "180000")
```

Comment:
This query result looks incorrect. I think [this](https://github.com/apache/incubator-druid/blob/master/processing/src/main/java/org/apache/druid/query/topn/NumericTopNMetricSpec.java#L108) also needs to be fixed.